ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,282 @@
+
+
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+
+    Returns:
+        pod_obj (obj): Registry pod obj
+
+    Raises:
+        UnexpectedBehaviour: When image-registry pod is not present.
+    """
+    # Sometimes when there is a update in config crd, there will be 2 registry pods
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    iteration = 0
+    while True:
+        iteration += 1
+        pod_data = pod.get_pods_having_label(
+            label='docker-registry=default', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+        )
+        pod_obj = [pod.Pod(**data) for data in pod_data]
+        if len(pod_obj) == 1:
+            break
+        elif len(pod_obj) == 0:
+            raise UnexpectedBehaviour("Image-registry pod not present")
+        elif iteration > 6:
+            raise UnexpectedBehaviour("Waited for 3 mins Image-registry pod is not in Running state")
+        else:
+            logger.info(f"Waiting for 30 sec's for registry pod to be up iteration {iteration}")
+            time.sleep(wait_time)
+    return pod_obj
+
+
+def get_oc_podman_login_cmd():
+    """
+    Function to get oc and podman login commands on node
+
+    Returns:
+        cmd_list (list): List of cmd for oc/podman login
+    """
+    user = config.RUN['username']
+    filename = os.path.join(
+        config.ENV_DATA['cluster_path'],
+        config.RUN['password_location']
+    )
+    with open(filename) as f:
+        password = f.read()
+    helpers.refresh_oc_login_connection()
+    ocp_obj = ocp.OCP()
+    token = ocp_obj.get_user_token()
+    route = get_default_route_name()
+    cmd_list = [
+        'export KUBECONFIG=/home/core/auth/kubeconfig',
+        f"oc login -u {user} -p {password}",
+        f"podman login {route} -u {user} -p {token}"
+    ]
+    master_list = helpers.get_master_nodes()
+    helpers.rsync_kubeconf_to_node(node=master_list[0])
+    return cmd_list
+
+
+def validate_pvc_mount_on_registry_pod():
+    """
+    Function to validate pvc mounted on the registry pod
+
+    Raises:
+        AssertionError: When PVC mount not present in the registry pod
+    """
+    pod_obj = get_registry_pod_obj()
+    mount_point = pod_obj[0].exec_cmd_on_pod(command="mount")
+    assert "/registry" in mount_point, f"pvc is not mounted on pod {pod_obj.name}"
+    logger.info("Verified pvc is mounted on image-registry pod")
+
+
+def validate_registry_pod_status():
+    """
+    Function to validate registry pod status
+    """
+    pod_obj = get_registry_pod_obj()
+    helpers.wait_for_resource_state(pod_obj[0], state=constants.STATUS_RUNNING)
+
+
+def validate_all_image_registry_pod_status():
+    """
+    Function to validate all image-registry pod status
+    """
+    pass
+
+
+def validate_data_at_backend():
+    """
+    Function to validate image_push data from ceph
+    """
+    pass
```

Comment:
Raise NotImplementedError when it's not implemented yet ore just remove this function from this PR

Reply:
ack